cmd/add_target.go

```diff
@@ -0,0 +1,786 @@
+
+			pterm.Println()
+
+			if selectedIndex == 0 {
+				break
+			}
+
+			param := optionalParameters[selectedIndex-1]
+
+			mask := strings.Contains(param.Name, "passw") || strings.Contains(param.Name, "secret")
+			value := textInput("Provide the value for the optional parameter "+pterm.Bold.Sprint(param.Name)+": "+param.Description, "Value for "+param.Name, "", mask)
+
+			addStringNodeWithValue(param.Name, value, targetNode)
+		}
+	}
+
+	return targetNode, targetName
+}
+
+func addStringNodeWithValue(key, value string, parent *ast.MappingNode) {
+	node := addStringNode(key, parent)
+	node.Value = value
+}
+
+func fetchPluginInfo(connector string) *pl.PluginInfo {
+	spinner, _ := pterm.DefaultSpinner.Start("Looking for connector information...")
+
+	client, err := plugin.NewPluginClient(connector, "", hclog.L())
+	if err != nil {
+		spinner.Fail(fmt.Sprintf("Unable to load connector %q: %s", connector, err.Error()))
+		pterm.Println()
+
+		return nil
+	}
+
+	defer client.Close()
+
+	info, err := client.GetInfo()
+	if err != nil {
+		spinner.Fail(fmt.Sprintf("Connector (%s) does not implement the Info interface", connector))
+		pterm.Println()
+
+		return nil
+	}
+
+	pluginInfo, err := info.GetInfo(context.Background())
+	if err != nil {
+		spinner.Fail(fmt.Sprintf("Unable to load plugin information: %s", err))
+		pterm.Println()
+
+		return nil
+	}
+
+	spinner.Success(fmt.Sprintf("Loaded connector information for %s", pluginInfo.Name))
+	pterm.Println()
+
+	return pluginInfo
+}
+
+func fetchDataSources(domain, apiUser, apiSecret string) map[string]DataSourceInfo {
+	spinner, _ := pterm.DefaultSpinner.Start("Loading available data sources from Raito Cloud...")
+
+	gql := `{ "operationName": "dataSources", "variables":{}, "query": "query dataSources {
+        dataSources {
+          ... on PagedResult {
+            edges {
+              node {
+                ... on DataSource {
+                  name
+                  id
+                  identityStores {
+                    id
+                    native
+                  }
+                }
+              }
+            }
+          }
+        }
+      }"}`
+
+	response := DataSourcesResponse{}
+	res, err := executeGraphQLWithCustomConfig(gql, domain, apiUser, apiSecret, &response)
+
+	if err != nil {
+		spinner.Fail("Unable to fetch data sources: " + err.Error())
+		pterm.Println()
+
+		return nil
+	}
+
+	if len(res.Errors) > 0 {
+		spinner.Fail("Unable to fetch data sources: " + res.Errors[0].Message)
+		pterm.Println()
+
+		return nil
+	}
+
+	spinner.Success("Data sources fetched successfully")
+	pterm.Println()
+
+	return response.GetDataSources()
+}
+
+func createDataSource(dsName, domain, apiUser, apiSecret string) *DataSourceInfo {
+	spinner, _ := pterm.DefaultSpinner.Start("Creating new data source in Raito Cloud...")
+
+	gql := fmt.Sprintf(`{ "operationName": "dataSource", "variables":{}, "query": "mutation dataSource {
+        createDataSource(input: { name: \"%s\" }) {
```

Comment:
I would prefer to use variables here. Especially because this could be set by the end user. Otherwise gql injection would be possible. 
It would maybe an option to use the `hasura` graphql client that is also defined in the CLI. You may want to have a look at https://github.com/raito-io/cli/blob/main/internal/data_source/metadata.go#L64 or https://github.com/raito-io/cli/blob/main/internal/version_management/version_management.go#L179.